### PR TITLE
Google Analytics and Facebook Pixel support

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | tba
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR brings:

- New Tracking settings tab in Configurations
- Google Analytics integration with USERID feature (https://support.google.com/analytics/answer/3123662?hl=en)
- Facebook Pixel integration with Advanced Matching (https://developers.facebook.com/ads/blog/post/2016/05/31/advanced-matching-pixel/)
- New Campaign action Send tracking event (usable just after  Visits a page decision)

Tracking should use for remarketing in Facebook Ads and Google Adwords. 

image

image

#### Steps to test this PR:

**Facebook Pixel** 

1.  Create Facebook pixel https://www.facebook.com/business/help/952192354843755
2.  Go to Configurations and add Facebook  Pixel ID end enable feature on landing/tracking page.
3. Install Facebook Pixel Helper https://chrome.google.com/webstore/detail/facebook-pixel-helper/fdgfkebogiimcoedlicjlajpkdmockpc
4. Go to tracking page and check Facebook Pixel Helper - 1 event should fire in debug window
5. Create campaign with page visit decision and add new action - Send tracking event
6. Add yourself to campaign, go to tracking page  and check Facebook Pixel Helper - 2 events should fired in debug window (see image below)

image

**Google Analytics** 

1.  Go to Configurations and add Google Analytics ID end enable feature on landing/tracking page
2. Install Tag assistent and enable recording on your website https://chrome.google.com/webstore/detail/tag-assistant-by-google/kejbdjndbnbjgmefkgdddjlbokphdefk
3. Go to tracking page and Tag assistent - 1 pageview request should fire in debug window
4. Create campaign with page visit decision and add  action - Send tracking event
5.  Add yourself to campaign, go to tracking page  and check Tag assistent - 1 pageview request and 1 events should fire in debug window (see image below)

image

### CODE REVIEW

**Comment by kuzmany** [comment]
@escopecz Ready for tests

**Comment by luizeof** [comment]
+1

\o/ Mautic Will Work like an tag manager, but campaign based !!!

GREAT Work @kuzmany

**Comment by luizeof** [comment]
Can you show an image of campaign action for fb pixel? Thanks

**Comment by kuzmany** [comment]
@escopecz please mark this PR as WIP
I decided it rework with GA events, Facebook Pixel and Google Adwords support and possible extend via plugins with another tracking service

File: app/bundles/PageBundle/PageEvents.php [hunk 1]
```diff
@@ -28,15 +28,24 @@
     const VIDEO_ON_HIT = 'mautic.video_on_hit';
 
     /**
-     * The mautic.page_on_hit event is thrown when a public page is browsed and a hit recorded in the analytics table.
-     *
-     * The event listener receives a Mautic\PageBundle\Event\PageHitEvent instance.
-     *
-     * @var string
-     */
+ * The mautic.page_on_hit event is thrown when a public page is browsed and a hit recorded in the analytics table.
```
Comment (alanhartless) [inline app/bundles/PageBundle/PageEvents.php]: Minor thing but please fix these comment indentions.

File: app/bundles/PageBundle/PageEvents.php [hunk 2]
```diff
@@ -28,15 +28,24 @@
...
+ *
+ * The event listener receives a Mautic\PageBundle\Event\PageHitEvent instance.
+ *
+ * @var string
+ */
     const PAGE_ON_HIT = 'mautic.page_on_hit';
 
     /**
+ * The mautic.track_on_hit event is thrown when a public page is browsed and a 3rd party tracking pixel is available
```
Comment (alanhartless) [inline app/bundles/PageBundle/PageEvents.php]: Minor thing but please fix these comment indentions.

**Comment by kuzmany** [comment]
@escopecz ready for tests ;)

**Comment by escopecz** [comment]
If I insert the Facebook Pixel ID to the Mautic configuration and save it, it's blank again after refresh. None of the switches for the Facebook Pixel keeps the value after save either. Google Analytics as well.

**Comment by kuzmany** [comment]
@escopecz try now

**Comment by escopecz** [comment]
I tested FB pixel and it works nicely, but isn't there the category missing in the campaign action? This is how is the campaign tracking event configured:
screen shot 2017-08-23 at 13 48 42
And this is what was sent to FB:
screen shot 2017-08-23 at 13 48 25
I can see action, label, but not category.

**Comment by kuzmany** [comment]
@escopecz 
Facebook pixel doesn't have category. Just action and label, category has tooltip with info.
I didn't want create two different actions but we can divided it.
Your FB pixel looks OK.

Facebook: action and label
GA: category, action and label

**Comment by escopecz** [comment]
Thanks for explanation. The tooltip translation is broken:
edit campaign mautic

**Comment by kuzmany** [comment]
@escopecz updated. 
The reason is we could add in future to tracking campaign action another  services like onesignal, adwords etc. Most of them has action and label, or category, action and label.

**Pull request merged**